server/gokbg3/grails-app/services/com/k_int/ConcurrencyManagerService.groovy

```diff
@@ -328,32 +298,46 @@ class ConcurrencyManagerService {
    * @param offset
    * @return List of Jobs
    */
-  public Map getGroupJobs(int group_id, int max = 10, int offset = 0) {
+  public Map getGroupJobs(def group_id, def max = 10, def offset = 0) {
+    return getFilteredJobs("group_id", group_id, max, offset)
+  }
+
+/**
+ * filters the job list for specific ids in named parameter fields.
+ * @param parameterName
+ * @param id
+ * @param max
+ * @param offset
+ * @return List of Jobs
+ */
+  private Map getFilteredJobs(String propertyName, def id, def max = 10, def offset = 0) {
     def allJobs = getJobs()
     def selected = []
     def result = [:]
     def total = null
 
-    if (group_id == null) {
+    if (id == null || propertyName == null) {
       return null
     }
 
-    log.debug("Getting jobs for group ${group_id}")
+    log.debug("Getting jobs for $paramName ${id}")
 
-    // Get the jobs.
+    // Filter the jobs.
     allJobs.each { k, v ->
-      if (v.groupId == group_id) {
+      if (v[propertyName] == id || v[propertyName]?.id == id) {
```

Comment:
This has to be done differently since for unmatched jobs from `groupId` & `ownerId` `v[propertyName].id` is undefined.

Reply:
Still getting an `MissingPropertyException: No such property: id for class: java.lang.Integer.`

Reply:
checking for the class of v[properyName] before using it